### PR TITLE
arrow-box component

### PR DESCRIPTION
**Pull request by jzhang300**
* add arrow-box component

### CODE REVIEW

File: example/src/components/ArrowBoxExamples.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    };
+  },
+
+  onClick1() {
+    this.setState({
+      box1Show: !this.state.box1Show,
+    });
+  },
+
+  onClick2() {
```
Comment (kognate) [inline example/src/components/ArrowBoxExamples.js:23]: this is a nitpick but I think the names of these click functions should be more clear or it should be one click function.  the box1 box2 thing is also a little shakey.

**Review by kognate — APPROVED** [review]

**Comment by nfriedly** [comment]
It looks like there's an issue with the z-index:

screen shot 2016-11-16 at 2 45 55 pm

(This is in firefox, haven't tested other browsers)

File: example/src/components/ArrowBoxExamples.js (hunk 1 above)
Comment (nfriedly) [reply example/src/components/ArrowBoxExamples.js:23]: Yea, it feels a bit off to me. I might just suggest having complete code in the first example and omitting the second one. 

If you wanted to go even farther, you could make the example stateless:

``` jsx
<div style={{ position: 'relative' }}>
  This is the element that the ArrowBox is anchored to.
  <ArrowBox
    direction="top"
    show={true}
    color={Colors.red_50}
    icon="error"
  >
    <p className="base--p" style={{ color: '#fff', textAlign: 'left' }}>
      This arrow box is now oriented to the <b>top</b>.
    </p>
  </ArrowBox>
</div>
```

File: example/src/components/ArrowBoxExamples.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import ArrowBox from '../../../src/components/ArrowBox';
+import Code from '../../../src/components/Code';
+import Colors from '../../../src/components/Colors';
```
Comment (nfriedly) [inline example/src/components/ArrowBoxExamples.js:4]: This should be shown in the example code (or else we should just use hex values)

**Review by nfriedly — CHANGES_REQUESTED** [review]
I think the z-index thing is probably a blocker, the rest is nice-to-have.

**Comment by nfriedly** [comment]
Given that the changes here are necessarily included in #29, I'd recommend closing this PR and just working on that one.

File: example/src/components/ArrowBoxExamples.js (hunk 1 above)
Comment (jzhang300) [reply example/src/components/ArrowBoxExamples.js:23]: Maybe the example was a little fancy :P

**Comment by jzhang300** [comment]
Closing this pr and carrying any changes to arrowbox over to #29

**Pull request closed without merging**